src/subs.jl

```diff
@@ -102,7 +102,7 @@ Convert a SymEngine numeric value into a number
 
 """
 N(b::Basic) = N(BasicType(b))
-N(b::BasicType{Val{:Integer}}) = eval(parse(toString(b)))  ## HACKY
+N(b::BasicNumber) = eval(parse(toString(b)))  ## HACKY
```

Comment:
Yes, For `RealDouble`s we can use `real_double_get_d`. Also there is `basic_eval` that converts Basic values to numeric types in SymEngine. (`RealDouble` and `ComplexDouble`)